Tune ORTModule logging experience a bit (#18298)
microsoft/onnxruntime: docs/ORTModule_Training_Guidelines.md

```
Check [DebugOptions implementation](../orttraining/orttraining/python/training/ortmodule/options.py) for more details.

> #### Log Level Explainations

Reviewer: ```suggestion
#### Log Level Explanations
```

<table>
<tr>
<th>Log Level</th>
<th>Description</th>
</tr>
<tr>

> <td>
> <div>FATAL | ERROR | WARNING (For Users)</div>
> <br>
> <div>`WARNING` is default and recommended level for users to use.</div>
> </td>

Reviewer: ```suggestion
<td>

`FATAL` | `ERROR` | `WARNING` (For Users)

`WARNING` is the default and recommended level for users to use.
</td>
```

<td>

- ONNX Runtime backend log level - `FATAL` | `ERROR` | `WARNING`.
- ORTModule log level - `FATAL` | `ERROR` | `WARNING`.
- Rank-0 log filtering is `ON` (e.g. logging on rank-0-only).
- PyTorch exporter export logs filtering is `ON`.
- PyTorch exporter verbose logs (including tracing graph) filtering is `ON`.

</td>
</tr>

<tr>

> <td>INFO (For Users | ORT Developers)</td>

Reviewer: ```suggestion
<td>

`INFO` (For Users | ORT Developers)
</td>
```

<td>

- ONNX Runtime backend log level - `WARNING`.
- ORTModule log level - `INFO`.
- Rank-0 log filtering is `ON` (e.g. logging on rank-0-only).
- PyTorch exporter export logs filtering is `ON`.
- PyTorch exporter verbose logs (including tracing graph) filtering is `OFF`.

</td>
</tr>

<tr>

> <td>DEVINFO (For ORT Developers)</td>

Reviewer: ```suggestion
<td>

`DEVINFO` (For ORT Developers)</td>
```

<td>

- ONNX Runtime backend `INFO` log level.
- ORTModule `INFO` log level.
- Rank-0 log filtering is `OFF` (e.g. logging on all ranks).
- PyTorch exporter export logs filtering is `OFF`.
- PyTorch exporter verbose logs (including tracing graph) filtering is `OFF`.

</td>
</tr>

<tr>

> <td>VERBOSE (For ORT Developers)</td>

Reviewer: ```suggestion
<td>

`VERBOSE` (For ORT Developers)</td>
```
